sql/connect/client/jdbc/src/main/scala/org/apache/spark/sql/connect/client/jdbc/SparkConnectResultSet.scala

```diff
@@ -501,28 +516,22 @@ class SparkConnectResultSet(
     throw new SQLFeatureNotSupportedException
 
   override def getDate(columnIndex: Int, cal: Calendar): Date = {
-    val date = getDate(columnIndex)
-    if (date == null || cal == null) {
-      return date
-    }
-
-    val targetCalendar = cal.clone().asInstanceOf[Calendar]
-    targetCalendar.setTime(date)
-    targetCalendar.set(Calendar.HOUR_OF_DAY, 0)
-    targetCalendar.set(Calendar.MINUTE, 0)
-    targetCalendar.set(Calendar.SECOND, 0)
-    targetCalendar.set(Calendar.MILLISECOND, 0)
-    new Date(targetCalendar.getTimeInMillis)
+    // The Calendar parameter in JDBC methods is meant for timezone-aware types,
```

Comment:
I suggest turning it into class or method comments so that it can be reflected in the Javadoc.